server.go

```diff
@@ -259,7 +302,48 @@ func (h *handler) processUnaryRpc(
 	}
 	defer cancel()
 
+	var appErr error
 	fullMethod := fmt.Sprintf("/%s/%s", info.name, md.MethodName)
+
+	var statsBegin *stats.Begin
+	incomingMetadata, imOk := metadata.FromIncomingContext(ctx)
+	if !imOk {
+		incomingMetadata = metadata.MD{}
+	}
+	for _, sh := range h.srv.statsHandlers {
+		ctx = sh.TagRPC(ctx, &stats.RPCTagInfo{
+			FullMethodName: fullMethod,
+		})
+
+		beginTime := time.Now()
+		statsBegin = &stats.Begin{
+			BeginTime:      beginTime,
+			IsClientStream: false,
+			IsServerStream: false,
+		}
+		sh.HandleRPC(ctx, statsBegin)
+		sh.HandleRPC(ctx, &stats.InHeader{
+			FullMethod: fullMethod,
+			//RemoteAddr:  t.Peer().Addr,
+			//LocalAddr:   t.Peer().LocalAddr,
+			//Compression: stream.RecvCompress(),
+			//WireLength:  stream.HeaderWireLength(),
+			Header: incomingMetadata,
+		})
+	}
+	defer func() {
+		for _, sh := range h.srv.statsHandlers {
```

Comment:
In the name of DRY, this block of code is used often so could be made into a util.